### PR TITLE
RSDEV-1084: Fix stoichiometry import revision — post-import Envers fixup

### PR DESCRIPTION
**Pull request by nhanlon2**
After XML archive import, stoichiometry references in field HTML had revision=null because Envers audit records are not available during the import transaction. Frontend now requires revision to be a number, causing broken stoichiometry display after import.

Add StoichiometryImportRevisionFixupManager that runs after import transaction commits, queries Envers for actual revisions, and rewrites field HTML with correct revision numbers.

- Create interface + impl in com.researchspace.service.archive
- Wire into ExportImportImpl.doImport() after archiveImporter returns
- Add 8 unit tests covering all paths (null fixup, multi-stoich, Envers failure, existing revision skip)
- Update integration test to assert non-null revision from DB

## Description ##
***REQUIRED***
- ***brief description of the change***
- ***don't repeat too much information that's already included in the jira/github issue***
- ***if the change includes any UI updates, include screenshots or screen recordings if you think it adds value***

## Design decisions
***OPTIONAL***
- ***if there were any particular design decisions made, list them here***

## Testing notes
***OPTIONAL***
- ***if there are any particular pre-requisites or instructions for manual testing, list them here***

### CODE REVIEW

**Comment by nhanlon2** [comment]
Testing manually for correctness and ONLY copilot assigned to review until verified

File: src/main/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import org.springframework.stereotype.Service;
+
+@Service("stoichiometryImportRevisionFixupManager")
+public class StoichiometryImportRevisionFixupManagerImpl
+    implements StoichiometryImportRevisionFixupManager {
+
+  private static final Logger log =
+      LoggerFactory.getLogger(StoichiometryImportRevisionFixupManagerImpl.class);
+
+  private static final String STOICHIOMETRY_ATTR = "data-stoichiometry-table";
```
Comment (rs-fraser) [inline src/main/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerImpl.java:27]: As of last week there's a new front-end `data-stoichiometry-table-only` (holding "empty" stoich tables that aren't attached to a chemical) attribute which will also need handled here (and anywhere else `data-stoichiometry-table` appears). Probably best as a separate ticket though.

File: src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  @Mock private AuditManager auditManager;
+  @Mock private FieldManager fieldManager;
+  @Mock private User user;
+  @Mock private ImportArchiveReport report;
+
+  @InjectMocks private StoichiometryImportRevisionFixupManagerImpl testee;
+
+  @BeforeEach
+  void setUp() {
+    Mockito.lenient().when(report.isSuccessful()).thenReturn(true);
```
Comment (rs-fraser) [inline src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java]: I don't think this is used. When I remove the `lenient()` all 8 tests fail with `UnnecessaryStubbingException`

File: src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+  private StructuredDocument mockStructuredDocument(long id) {
+    StructuredDocument doc = Mockito.mock(StructuredDocument.class);
+    when(doc.getId()).thenReturn(id);
+    return doc;
+  }
+
+  @Test
+  void whenReportNotSuccessfulStillDoesFixup() {
+    Mockito.lenient().when(report.isSuccessful()).thenReturn(false);
```
Comment (rs-fraser) [inline src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java]: this line's not needed.

**Review by rs-fraser — APPROVED** [review]
Looks good. As discussed we now need to handle the new `data-stoichiometry-table-only` attribute but will handle that separately. Also added a couple comments about `lenient`

File: src/main/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerImpl.java (hunk 1 above)
Comment (nhanlon2) [reply src/main/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerImpl.java:27]: Yes - RSdev-1091 is handling this and further tickets may also be needed after that is done.

File: src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java (hunk 2 above)
Comment (nhanlon2) [reply src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java]: removed

File: src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java (hunk 3 above)
Comment (nhanlon2) [reply src/test/java/com/researchspace/service/archive/StoichiometryImportRevisionFixupManagerTest.java]: removed
